### PR TITLE
[1LP][RFR] Automation/Ansible/Credentials page model

### PR DESCRIPTION
**Pull request by quarckster**
Intent
=================

Automation/Ansible/Credentials page model and respectful crud test created

{{pytest: -v -k "test_embedded_ansible_credential_crud" --long-running}}

### CODE REVIEW

File: cfme/ansible/credentials.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+        )
+
+    def before_fill(self, values):
+        for name in self.widget_names:
+            if name not in values or values[name] is None:
+                continue
+            widget = getattr(self, name)
+            title = self.browser.get_attribute("title", widget)
+            try:
+                self.input(title).enable
```
Comment (izapolsk) [inline cfme/ansible/credentials.py]: :+1: but fix it. it should be enable()

File: cfme/ansible/credentials.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+            "Amazon": amazon_credential_fill_dict,
+            "VMware": vmware_credential_fill_dict
+        }
+
+        add_page.fill({"name": name, "credential_type": credential_type})
+        add_page.credential_form.fill(credential_type_map[credential_type])
+        add_page.add_button.click()
+        credentials_list_page = self.create_view(CredentialsListView)
+        # Without this StaleElementReferenceException can be raised
+        wait_for(lambda: False, silent_failure=True, timeout=5)
```
Comment (izapolsk) [inline cfme/ansible/credentials.py]: can you please give more explanation about this case ?

File: cfme/ansible/credentials.py [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+    @credential_form.register("Machine")
+    class CredentialFormMachineView(View):
+        username = Input(locator='.//input[@title="Username for this credential"]')
+        password = Input(locator='.//input[@title="Password for this credential"][2]')
+        private_key = TextInput(
+            locator='.//textarea[@title="RSA or DSA private key to be used instead of password"][2]'
+        )
+        private_key_phrase = Input(
+            locator='.//input[@title="Passphrase to unlock SSH private key if encrypted"][2]')
+        privilage_escalation = BootstrapSelect("{{name}}")
```
Comment (izapolsk) [inline cfme/ansible/credentials.py]: yeah, it really has such name. I suspect this has to be resolved to appropriate name. Don't you want to submit a ticket ?

Comment (quarckster) [inline cfme/ansible/credentials.py]: @izapolsk Absolutely

File: cfme/ansible/credentials.py (hunk 2 above)
Comment (quarckster) [inline cfme/ansible/credentials.py]: I encountered with sporadic `StaleElementReferenceException` without waiting a while after page is loaded. I can't explain the reason why it happens. I suspect some js magic is there.

File: cfme/ansible/credentials.py (hunk 3 above)
Comment (mfalesni) [inline cfme/ansible/credentials.py]: Looks like Angular :)

Comment (izapolsk) [inline cfme/ansible/credentials.py]: it looks to me like not resolved jinja variable :)

**Comment by izapolsk** [comment]
@quarckster, btw, please pay attention on exception in 5.8 PRT

**Pull request merged**